modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/api/construction/ConstructionParse.scala

```diff
@@ -0,0 +1,24 @@
+package org.tessellation.rosetta.domain.api.construction
+
+import cats.data.NonEmptyList
+
+import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
+import org.tessellation.rosetta.domain.operation.Operation
+import org.tessellation.rosetta.domain.{AccountIdentifier, NetworkIdentifier}
+import org.tessellation.security.hex.Hex
+
+import derevo.circe.magnolia.{customizableDecoder, customizableEncoder}
+import derevo.derive
+
+case object ConstructionParse {
+  @derive(customizableDecoder)
+  case class Request(networkIdentifier: NetworkIdentifier, signed: Boolean, transaction: Hex)
+
+  @derive(customizableEncoder)
+  case class Response(operations: List[Operation], accountIdentifierSigners: Option[NonEmptyList[AccountIdentifier]])
+
+  case class ServiceReturn(
+    operations: NonEmptyList[Operation],
+    accountIdentifierSigners: Option[NonEmptyList[AccountIdentifier]]
+  )
```

Comment:
imo it's totally fine to return a tuple from the service. `ServiceReturn` is just a bad name and I cannot think of any better one 🤔

Reply:
I mean, this idea is brillant if you can find a better name (actually the name that describes what it is, instead of where it is used). Otherwise, just return a tuple

Reply:
Renamed to `ParseResult`.